providers/core.tcpsocket.js

```diff
@@ -75,15 +75,10 @@ Socket_chrome.prototype.connect = function(hostname, port, cb) {
   chrome.sockets.tcp.create({}, function(createInfo) {
     this.id = createInfo.socketId;
     try {
-      chrome.sockets.tcp.connect(this.id, hostname, port, function (result) {
-        if (result < 0) {
-          var errorString = Socket_chrome.errorStringOfCode(result);
-          var freedomErrorCode = Socket_chrome.freedomErrorCode(errorString);
-          cb(undefined, {
-            'errcode': freedomErrorCode,
-            'message': 'Chrome Connection Failed: ' + errorString +
-                ' ' + chrome.runtime.lastError.message
-          });
+      chrome.sockets.tcp.connect(this.id, hostname, port, function(result) {
+        var error = Socket_chrome.chromeErrorHandler(result);
```

Comment:
Thanks for looking at this!

I'm not sure this is quite right, though. Look at what we had to do for `send`:
https://github.com/freedomjs/freedom-for-chrome/blob/master/providers/core.tcpsocket.js#L229

Basically, you have to examine `chrome.runtime.lastError` in addition to checking the return value. Does that make sense?